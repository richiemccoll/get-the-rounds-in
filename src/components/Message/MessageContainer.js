import React, { Component } from "react";
import { Message } from "semantic-ui-react";
import PropTypes from 'prop-types';

class MessageContainer extends Component {
  constructor(props) {
    super(props);
    this.handleDismiss = this.handleDismiss.bind(this);
  }

  componentDidMount() {
    setTimeout(() => this.props.hideNotification(), 3000);
  }

  handleDismiss() {
    this.props.hideNotification();
  }

  render() {
    return (
      <Message
        success
        content={this.props.message}
        onDismiss={this.handleDismiss}
      />
    );
  }
}

export default MessageContainer;

MessageContainer.PropTypes = {
    message: PropTypes.string,
    hideNotification: PropTypes.func.isRequired
}