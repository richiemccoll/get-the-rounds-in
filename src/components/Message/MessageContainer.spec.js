import React from "react";
import MessageContainer from "./MessageContainer";
import renderer from "react-test-renderer";

describe("Message Container", () => {
  const message = {
    message: "This is a notficiation",
    visible: true
  };

  const hideNotification = () => console.log("hiding notification");

  it("should render a message notification", () => {
    const component = renderer.create(
      <MessageContainer message={message.message} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
