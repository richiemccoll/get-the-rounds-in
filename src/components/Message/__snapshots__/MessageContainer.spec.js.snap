// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Container should render a message notification 1`] = `
<div
  className="ui success message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="content"
  >
    <p>
      This is a notficiation
    </p>
  </div>
</div>
`;
